R/formatter.r

```diff
@@ -74,32 +76,151 @@ number <- function(x, accuracy = 1, scale = 1, prefix = "",
   ret
 }
 
-#' Comma formatter: format number with commas separating thousands.
-#'
-#' @param ... other arguments passed on to [format()]
-#' @param x a numeric vector to format
-#' @return a function with single parameter x, a numeric vector, that
-#'   returns a character vector
+precision <- function(x) {
+  if (all(is.infinite(x))) {
+    return(1)
+  }
+
+  rng <- range(x, na.rm = TRUE, finite = TRUE)
+
+  span <- if (zero_range(rng)) abs(rng[1]) else diff(rng)
+  if (span == 0) {
+    return(1)
+  }
+
+  10^floor(log10(span))
+}
+
 #' @export
+#' @rdname number_format
+#' @param digits Deprecated, use `accuracy` instead.
 #' @examples
+#' # comma() and comma_format()
 #' comma_format()(c(1, 1e3, 2000, 1e6))
-#' comma_format(digits = 9)(c(1, 1e3, 2000, 1e6))
+#' comma_format(accuracy = .01)(c(1, 1e3, 2000, 1e6))
 #' comma(c(1, 1e3, 2000, 1e6))
 #'
-#' # If you're European you can switch . and , with the more general
-#' # format_format
-#' point <- format_format(big.mark = ".", decimal.mark = ",", scientific = FALSE)
-#' point(c(1, 1e3, 2000, 1e6))
-#' point(c(1, 1.021, 1000.01))
-comma_format <- function(...) {
-  force_all(...)
-  function(x) comma(x, ...)
+comma_format <- function(accuracy = 1, scale = 1, prefix = "",
+                         suffix = "", big.mark = ",", decimal.mark = ".",
+                         trim = TRUE, digits, ...) {
+  if (!missing(digits)) {
+    warning("'digits' argument is deprecated, use 'accuracy' instead.")
```

Comment:
Needs `call. = FALSE`, and instead of `'digits'` can you please do `` `digits` ``? (And similarly for accuracy)

Reply:
done